### PR TITLE
New page: Deployment Considerations (for services)

### PR DESCRIPTION
**Pull request by mmattel**
Fixes: #300 (document service limits (HA / scalability) )

Add a new page covering deployment considerations for services.

@wkloucek pls have a look. Note that we can always add/change stuff but at least we have started.

### CODE REVIEW

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+== Scaling of Services
+
+Though the majority of services can be scaled by starting them multiple times, this is not always possible for some of them mostly due to technical limitations when targeting High Availability or scaled installations.
+
+* The xref:{s-path}/search.adoc[Search service] can not be started multiple times. The use in HA or scaled installations is therfore limited. There is for the time being no replacement planned.
+
+* The xref:{s-path}/idm.adoc[IDM service] cannot be started multiple times and should not be used in HA / scaled installations, but replaced instead by an LDAP server that supports HA / scaling. Even if no HA / scaling is needed, the IDM service is only supported for a limited amount of users. Consider the limitation up to approximately 200 users.
+
+* The xref:{s-path}/idp.adoc[IDP service] cannot be started multiple times and should not be used in HA / scaled installations, but instead replaced by an OIDC provider that supports HA / scaling. The IDP service has a bare minimum feature set like you cannot easily revoke single sessions.
```
Comment (phil-davis) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:21]:
```suggestion
* The xref:{s-path}/idp.adoc[IDP service] cannot be started multiple times and should not be used in HA / scaled installations, but instead replaced by an OIDC provider that supports HA / scaling. The IDP service has a bare minimum feature set. For example, cannot easily revoke single sessions.
```

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+* The xref:{s-path}/store.adoc[Store service] cannot be started multiple times. ownCloud is looking for possibilities removing or making the store service able to scale.
```
Comment (phil-davis) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:23]:
```suggestion
* The xref:{s-path}/store.adoc[Store service] cannot be started multiple times. ownCloud is looking for ways of removing or making the store service able to scale.
```

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+* The xref:{s-path}/nats.adoc[NATS service] cannot be started multiple times and should not be used in HA or scaled installations. Instead it must be replaced by a NATS installation that is HA / scaled enabled. For details see the https://nats.io pages: {nats-clustering-url}[NATS Server Clustering] and {nats-helm-url}[Deploying NATS with Helm].
```
Comment (phil-davis) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:25]:
```suggestion
* The xref:{s-path}/nats.adoc[NATS service] cannot be started multiple times and should not be used in HA or scaled installations. Instead it must be replaced by a NATS installation that is HA / scaled enabled. For details see the https://nats.io pages: {nats-clustering-url}[NATS Server Clustering] and {nats-helm-url}[Deploying NATS with Helm] documentation.
```

**Review by phil-davis — APPROVED** [review]

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
+= Deployment Considerations
+:toc: right
+
+:nats-clustering-url: https://docs.nats.io/running-a-nats-service/configuration/clustering
+:nats-helm-url: https://docs.nats.io/running-a-nats-service/nats-kubernetes/helm-chart
+
+:description: When using container orchestration, some considerations should be taken when deploying services with respect to scaling.
```
Comment (EParzefall) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:7]:
```suggestion
:description: When using container orchestration, some considerations should be taken into account when deploying services with respect to scaling.
```

**Review by EParzefall — APPROVED** [review]

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+== Introduction
+
+{description}
+
+== Scaling of Services
+
+Though the majority of services can be scaled by starting them multiple times, this is not always possible for some of them mostly due to technical limitations when targeting High Availability or scaled installations.
+
+* The xref:{s-path}/search.adoc[Search service] cannot be started multiple times. The use in HA or scaled installations is therefore limited. There is, for the time being, no replacement planned.
```
[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc:17]: I was just rewording your text: `and there is no plugin replacement` from the referenced issue.
Happy to change this, just give me input...

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+* The xref:{s-path}/idm.adoc[IDM service] cannot be started multiple times and should not be used in HA / scaled installations, but replaced instead by an LDAP server that supports HA / scaling. Even if no HA / scaling is needed, the IDM service is only supported for a limited amount of users. Consider the limitation up to approximately 200 users.
```
[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc:19]: Can you clarify this for me - possibly before Christmas so we can fix and merge this?

[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc:19]: @dragotin maybe you have an idea if 200 fits.

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:19]:
```suggestion
* The xref:{s-path}/idm.adoc[IDM service] cannot be started multiple times and should not be used in HA / scaled installations, but replaced instead by an LDAP server that supports HA / scaling. If no LDAP server is available or no HA / scaling is needed, the IDM service can still be used in production environments but does by nature not cover all capabilities a professional LDAP server provides.
```
Relaxing the statement. I guess that a limiting number may never be sufficient but I strongly believe that any company that has more than a couple of users has in its backend an LDAP server. @dragotin what is your POV?

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc:17]: ok, will rephrase

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+== Introduction
+
+{description}
+
+== Scaling of Services
+
+Though the majority of services can be scaled by starting them multiple times, this is not always possible for some of them mostly due to technical limitations when targeting High Availability or scaled installations.
+
+* The xref:{s-path}/search.adoc[Search service] cannot be started multiple times. The use in HA or scaled installations is therefore limited. There is, for the time being, no replacement planned.
```
Comment (mmattel) [inline modules/ROOT/pages/deployment/services/deployment-considerations.adoc:17]:
```suggestion
* The xref:{s-path}/search.adoc[Search service] cannot be started multiple times. The use in HA or scaled installations is therefore limited.
```
Post a discussion with @micbar and @dragotin, the sentence about the future is removed.

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc]: fixed

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc (hunk 7 above)
Comment (dragotin) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc:19]: sounds good!

**Review by dragotin — APPROVED** [review]

File: modules/ROOT/pages/deployment/services/deployment-considerations.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (mmattel) [reply modules/ROOT/pages/deployment/services/deployment-considerations.adoc]: Suggestion committed as ok for @dragotin, resolving this point.

**Pull request merged**